File: Backend/ClearChain.Domain/Entities/Dispute.cs
namespace ClearChain.Domain.Entities;

public class Dispute
{
    public Guid Id { get; set; }
    public Guid PickupRequestId { get; set; }
    public Guid InitiatorId { get; set; }      // NGO who opens dispute
    public string Reason { get; set; } = string.Empty;   // wrong_items, wrong_quantity, damaged, other
    public string? NgoStatement { get; set; }
    public string? GroceryStatement { get; set; }
    public string? PhotoEvidenceUrl { get; set; }
    public string Status { get; set; } = "open";    // open, under_review, resolved_ngo, resolved_grocery, dismissed
    public string? AdminResolution { get; set; }
    public Guid? ResolvedByAdminId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? ResolvedAt { get; set; }

    public Organization? Initiator { get; set; }
    public PickupRequest? PickupRequest { get; set; }
}

File: Mobile/app/src/main/java/com/clearchain/app/domain/model/Dispute.kt
package com.clearchain.app.domain.model

data class Dispute(
    val id: String,
    val pickupRequestId: String,
    val initiatorId: String,
    val initiatorName: String,
    val reason: String,
    val ngoStatement: String?,
    val groceryStatement: String?,
    val photoEvidenceUrl: String?,
    val status: String,           // open, under_review, resolved_ngo, resolved_grocery, dismissed
    val adminResolution: String?,
    val createdAt: String,
    val resolvedAt: String?
) {
    val isResolved: Boolean get() = status.startsWith("resolved") || status == "dismissed"
    val isOpen: Boolean get() = status == "open"
    val isUnderReview: Boolean get() = status == "under_review"
}
